aeron-client/src/main/c/concurrent/aeron_thread.c

```diff
@@ -127,25 +138,41 @@ int aeron_thread_attr_init(pthread_attr_t *attr)
 static DWORD WINAPI aeron_thread_proc(LPVOID parameter)
 {
     aeron_thread_t *thread = (aeron_thread_t*)parameter;
-    thread->result = thread->callback(thread->arg0);
+    (*thread)->result = (*thread)->callback((*thread)->arg0);
     return 0;
 }
 
-int aeron_thread_create(aeron_thread_t *thread, void *attr, void*(*callback)(void*), void *arg0)
+int aeron_thread_create(aeron_thread_t *thread_ptr, void *attr, void*(*callback)(void*), void *arg0)
 {
+    if (thread_ptr == NULL)
+    {
+        return -1;
+    }
+    aeron_thread_t thread = (aeron_thread_t)malloc(sizeof(struct aeron_thread_stct));
```

Comment:
Do not use naked mallocs. Use `aeron_alloc`.

Reply:
resovled, also other malloc and free are resolved.